src/java/org/apache/cassandra/db/SystemKeyspace.java

```diff
@@ -1344,12 +1344,17 @@ public static void clearEstimates(String keyspace)
     public static synchronized SetMultimap<String, String> getTablesWithSizeEstimates()
     {
         SetMultimap<String, String> keyspaceTableMap = HashMultimap.create();
-        String cql = String.format("SELECT keyspace_name, table_name FROM %s", TableEstimates.toString(), TABLE_ESTIMATES_TYPE_PRIMARY);
-        UntypedResultSet rs = executeInternal(cql);
-        for (UntypedResultSet.Row row : rs)
+        // Its possible that size_estimates knows about a different set of keyspace/tables than table_estimates (mostly
+        // caused by external systems modifying the tables, such as dtest) so query both
+        for (String cql : Arrays.asList(
+            "SELECT keyspace_name, table_name FROM " + TableEstimates.toString(),
+            "SELECT keyspace_name, table_name FROM " + LegacySizeEstimates.toString()))
```

Comment:
what's the performance impact of going from one table scan to two here?

Reply:
I believe we have to do 2 table scans now until the table `size_estimates` goes away.

Reply:
nit: imo it would be more obvious to use `SchemaConstants.SYSTEM_KEYSPACE_NAME, TABLE_ESTIMATES` /  `SchemaConstants.SYSTEM_KEYSPACE_NAME, LEGACY_SIZE_ESTIMATES` in the constructed CQL query than using `TableEstimates.toString()` / `LegacySizeEstimates.toString()`.

Reply:
Do we expect the set of keyspaces in `table_estimates` and `size_estimates` to be mutually exclusive? If not, aren't you clobbering the estimates from `table_estimates` with the values from `size_estimates` table? Is that the expected behavior?

Reply:
@michaelsembwever the impact is related to the number of tables in the cluster; if you have 10 tables this is small, if you have 1k its much larger cost.  To @nastra point, its sadly unavoidable since the two tables could drift.  If you think about it from the migration standpoint, on T1 we have size_estimates as the source of truth, in T2 its now table_estimates, so if there happens to exist tables in size_estimates which are no longer in the new table_estimates they would live there forever.

The function is only called at two places:

1) CassandraDaemon startup.  This could impact startup times as its now twice the cost; impacts clusters with more tables; Sadly there is no monitoring around this, so only know via stack traces...
2) JMX. triggered via nodetool or via random JMX call.

Reply:
@nastra not sure I agree with the comment that toString is less obvious, is there a reason you say that?  In this case we don't care about quoting since it is known, but the `TableMetadata.toString` is a safer way to do this in general, so should be the new norm (in my opinion) rather than the common `format("%s.%s", ks_name, name)` (which is unsafe in general).

Reply:
> Do we expect the set of keyspaces in table_estimates and size_estimates to be mutually exclusive? If not, aren't you clobbering the estimates from table_estimates with the values from size_estimates table? Is that the expected behavior?

@dineshjoshi I expect them to conflict 99% of the time (1% for random users writing random data), which is why a set is used to dedup; after a reboot they should have identical tables.

>  clobbering the estimates from table_estimates with the values from size_estimates table?

This function only looks for table names, not values, so no clobbering estimates.

Reply:
I switched to truncate as this was a more expensive truncate....  That should resolve all comments another than the toString one.

Reply:
> @nastra not sure I agree with the comment that toString is less obvious, is there a reason you say that? In this case we don't care about quoting since it is known, but the `TableMetadata.toString` is a safer way to do this in general, so should be the new norm (in my opinion) rather than the common `format("%s.%s", ks_name, name)` (which is unsafe in general).

@dcapwell it's really just a personal preference and I don't have any strong objections around how it's done at the end. That's why I marked it as a nit. Your approach makes sense and is perfectly fine